### PR TITLE
Introduce --binary-customisation-script for inserting binary hooks

### PR DESCRIPTION
**Pull request by OddBloke**

### CODE REVIEW

**Comment by OddBloke** [comment]
I've confirmed that the yaml produced by this looks good, but I'm still waiting for an actual build run to happen; please review but leave for me to merge once I've confirmed that.

File: generate_build_config.py [hunk 1]
```diff
@@ -73,11 +86,14 @@ def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('output_filename')
     parser.add_argument('--customisation-script', dest='custom_script')
+    parser.add_argument('--binary-customisation-script',
+                        dest='binary_custom_script')
```
Comment (rcj4747) [inline generate_build_config.py]: README.md has not been updated to reflect the addition of argument flag parsing or these specific changes (with example/explanation of the two script types).

**Review by chrisglass — APPROVED** [review]
Looks good. The documentation needs to be changed, but that can be done in another branch.

**Review by rcj4747 — CHANGES_REQUESTED** [review]
I would like to see README.md updated to, at minimum, document the cmdline change.

File: generate_build_config.py (hunk 1 above)
Comment (OddBloke) [reply generate_build_config.py]: #14 handles fixing the argument parsing for `--customisation-script`.

Comment (OddBloke) [reply generate_build_config.py]: In terms of README updates, I'm worried about making that document too long or difficult to parse, as that's the introduction that users will have to the project.

Most users won't care about binary customisations, so I don't want to dilute the clear journey in the README.md.

That said, we do need to think about more detailed documentation, but I'd like to have that outside the scope of this PR.

(I should add a help message though, so let me do that.)

Comment (rcj4747) [reply generate_build_config.py]: Sounds good.  Help message on the argument and I'm +1

**Review by rcj4747 — APPROVED** [review]

**Pull request merged**